packages/ethereum-contracts/contracts/agreements/ConstantFlowAgreementV1.sol

```diff
@@ -813,6 +813,88 @@ contract ConstantFlowAgreementV1 is
         emit FlowOperatorUpdated(token, currentContext.msgSender, flowOperator, permissions, flowRateAllowance);
     }
 
+    /// @dev IConstantFlowAgreementV1.increaseFlowRateAllowance implementation
+    function increaseFlowRateAllowance(
+        ISuperfluidToken token,
+        address flowOperator,
+        int96 flowRateAllowanceDelta, // flowRateBudget
+        bytes calldata ctx
+    ) public override returns (bytes memory newCtx) {
+        newCtx = ctx;
+
+        // [SECURITY] NOTE: we are holding the assumption here that ctx is correct and we validate it with
+        // authorizeTokenAccess:
+        ISuperfluid.Context memory currentContext = AgreementLibrary.authorizeTokenAccess(token, ctx);
+
+        if (currentContext.msgSender == flowOperator) revert CFA_ACL_NO_SENDER_FLOW_OPERATOR();
+        if (flowRateAllowanceDelta < 0) revert CFA_ACL_NO_NEGATIVE_ALLOWANCE();
+        
+        (
+            bytes32 flowOperatorId,
+            uint8 oldPermissions,
+            int96 oldFlowRateAllowance
+        ) = getFlowOperatorData(token, currentContext.msgSender, flowOperator);
+
+        // @note this will revert if it overflows
+        int96 newFlowRateAllowance = oldFlowRateAllowance + flowRateAllowanceDelta;
+        _updateFlowRateAllowance(
+            token,
+            flowOperatorId,
+            oldPermissions,
+            newFlowRateAllowance
+        );
+
+        emit FlowOperatorUpdated(
+            token,
+            currentContext.msgSender,
+            flowOperator,
```

Comment:
Just a sanity check for myself: `flowOperator` (the address) and `flowOperatorId` (assuming something hashed together) are different things, right?

Reply:
yes, the `flowOperator` is the grantee of the permissions.
`flowOperatorId` is `keccak256(abi.encode("flowOperator", flowSender, flowOperator))`